crates/matrix-sdk-ui/CHANGELOG.md

```diff
@@ -24,6 +24,12 @@ All notable changes to this project will be documented in this file.
   implement `Into<PathBuf>` also implement `Into<AttachmentSource>`.
   ([#4451](https://github.com/matrix-org/matrix-rust-sdk/pull/4451))
 
+- [**breaking**] Add an "offline" mode to the `SyncService`. This allows the
+  `SyncService` to attempt to restart the sync automatically. It can be enabled
+  with the `SyncServiceBuilder::with_offline_mode` method. Due to this addition,
+  the `SyncService::start` method now returns a `Result` type.
```

Comment:
We touched offline on whether this is actually useful/actionable to the end client and we agreed that eventual (rare) errors can be handled internally here and the start/stop methods simplified.

Reply:
The `SyncService::stop()` method has been made infallible in https://github.com/matrix-org/matrix-rust-sdk/pull/4592/commits/bd77e273ecd8e6745db12828d083cd26f8181705.

Due to this, the `SyncService::start()` has become infallible as well.